aiida_kkr/tools/multi_imps_data_extract.py

```diff
@@ -0,0 +1,453 @@
+# -*- coding: utf-8 -*-
+
+"""Magnetic data collection.
+
+This module collect the magnetic data from <combine_imps_wc> node, when the 
+the <combine_imps_w> combines two single impurity wc <kkr_imp_wc> or one 
+<combine_imps_wc> and <one kkr_imp_wc>.
+
+Module includes two classes:
+    CoupleImpurityData: The class collects the data of a couple of impurities
+        from a specific <combine_imps_wc> node .
+
+    MultiImpuritiesData: The class collects the data from all possible couples 
+    of the <combine_imps_wc> by implmenting CoupleImpurityData class.
+    And it  repeats the same process for all the <combine_imps_wc> nodes given 
+    by python list.
+"""
+
+
+from __future__ import absolute_import
+from aiida.orm import Node
+from typing import List
+import math as m
+#from aiida_kkr.calculations import KkrimpCalculation
+from six.moves import range
+
+class CoupleImpurityData(object):
+    """A magnetic data of impurity couple.
+    
+    The class must be instantiated by combine_imps_wc 'node' and 'imp_orders'
+    dict. Where the 'imp_orders' dict includes the values like 0,1 or 1,2 etc
+    under any key name e.g. key1, key2. The numbers indicate the a couple
+    first-second (0,1) impurities, second-third (1,2) impurities respectively.
```

Comment:
does this index refer to the position in the big impurity cluster? Please clarify